src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.IncrementalAnalyzerProcessor.cs

```diff
@@ -187,13 +254,17 @@ private void ReportPendingWorkItemCount()
                 private async Task RunAnalyzersAsync<T>(
                     ImmutableArray<IIncrementalAnalyzer> analyzers,
                     T value,
+                    WorkItem? workItem,
                     Func<IIncrementalAnalyzer, T, CancellationToken, Task> runnerAsync,
                     CancellationToken cancellationToken)
                 {
```

Comment:
instead of making tihs optional, have two overloads.  this one takes a non-optional workitem and passes its applicable analyzers to the next overload.  

tehn, in teh one place i can see where you don't have a workitem, it just calls the overloads with the exact analyzers it wants.